### PR TITLE
Add retries to EventTracker's event posting

### PR DESCRIPTION
**Pull request by uzi**
We're seeing noise in the logs when EventTracker's
post fails.  Instead, retry two times with a
random delay and then give up silently.

👓  @schwers @phil303 @nramadas

### CODE REVIEW

**Comment by phil303** [comment]
Can you give us more context in the commit please

**Comment by uzi** [comment]
💇

File: src/lib/eventTracker.js [hunk 1]
```diff
@@ -15,14 +17,20 @@ function calculateHash(key, string) {
   return hmac.read();
 }
 
-function post({url, data, query, headers, done}) {
+function post({url, data, query, headers, done}, retry = 0) {
```
Comment (phil303) [inline src/lib/eventTracker.js]: nit, no spaces before/after equals

File: src/lib/eventTracker.js [hunk 2]
```diff
@@ -15,14 +17,20 @@ function calculateHash(key, string) {
...
     .post(url)
     .query(query)
     .set(headers)
     .timeout(DEFAULT_API_TIMEOUT)
     .send(data)
-    .then(done);
+    .then(done)
+    .catch(() => {
+      if (retry < POST_RETRIES) {
+        const delay = Math.floor(1000 + Math.random() * 1000);
```
Comment (phil303) [inline src/lib/eventTracker.js]: :no_bell: so always between 1 and 2 seconds? Any reasoning for that?

File: src/lib/eventTracker.js [hunk 3]
```diff
@@ -15,14 +17,20 @@ function calculateHash(key, string) {
...
   return makeRequest
     .post(url)
     .query(query)
     .set(headers)
     .timeout(DEFAULT_API_TIMEOUT)
     .send(data)
-    .then(done);
+    .then(done)
+    .catch(() => {
+      if (retry < POST_RETRIES) {
```
Comment (phil303) [inline src/lib/eventTracker.js]: should we log if this get exceeded? Might be able to tease out what's going on

**Comment by phil303** [comment]
:nail_care:

File: src/lib/eventTracker.js (hunk 2 above)
Comment (uzi) [inline src/lib/eventTracker.js]: Just used the same logic as session refreshing... granted the comment there is wrong:

```
    // retry, only once, somewhere between 1 and 3 seconds later
    setTimeout(() => dispatch(refresh()), Math.floor(1000 + Math.random() * 1000));
```

Didn't want it too long or too short, basically.

File: src/lib/eventTracker.js (hunk 1 above)
Comment (uzi) [inline src/lib/eventTracker.js]: Sure.

File: src/lib/eventTracker.js (hunk 3 above)
Comment (uzi) [inline src/lib/eventTracker.js]: There's not much to log.  I checked e and the only thing in there is the timeout value.  That's before I did this retry code.

Comment (phil303) [inline src/lib/eventTracker.js]: oh i dont know about that. we could log the data, headers, query params. There's a lot going on here other than just the error we get back. I'm wondering if there might be a malformed request causing the issue

**Comment by uzi** [comment]
Abandoning for now

**Pull request closed without merging**